Sources/PostgresNIO/New/PostgresCodable.swift

```diff
@@ -3,30 +3,69 @@ import class Foundation.JSONEncoder
 import class Foundation.JSONDecoder
 
 /// A type that can encode itself to a postgres wire binary representation.
-public protocol PostgresEncodable {
+public protocol PostgresEncodable: PostgresDynamicTypeEncodable {
     // TODO: Rename to `PostgresThrowingEncodable` with next major release
 
-    /// identifies the data type that we will encode into `byteBuffer` in `encode`
+    /// Identifies the data type that we will encode into `byteBuffer` in `encode`
     static var psqlType: PostgresDataType { get }
 
-    /// identifies the postgres format that is used to encode the value into `byteBuffer` in `encode`
+    /// Identifies the postgres format that is used to encode the value into `byteBuffer` in `encode`
     static var psqlFormat: PostgresFormat { get }
 
     /// Encode the entity into the `byteBuffer` in Postgres binary format, without setting
     /// the byte count. This method is called from the ``PostgresBindings``.
-    func encode<JSONEncoder: PostgresJSONEncoder>(into byteBuffer: inout ByteBuffer, context: PostgresEncodingContext<JSONEncoder>) throws
+    func encode<JSONEncoder: PostgresJSONEncoder>(
+        into byteBuffer: inout ByteBuffer,
+        context: PostgresEncodingContext<JSONEncoder>
+    ) throws
+}
+
+/// A type that can encode itself to a Postgres wire binary representation.
+/// Dynamic types are types that are not known at compile time. For example, a custom type that is created
+/// at runtime.
+public protocol PostgresDynamicTypeEncodable {
```

Comment:
This should be `PostgresDynamicTypeThrowingEncodable`

Reply:
For the transition phase (`// TODO: Rename to `PostgresThrowingEncodable` with next major release`)  this will be super confusing. So until the next major version `PostgresEncodable` will be the throwing version and `PostgresDynamicTypeEncodable` the non-throwing one, I'll have to put a mental note somewhere.
Tho I get the point 😆 Updated! ✅